File: Java/projet_Bike_Desktop/src/entites/Maitenance.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entites;

import java.util.Date;

/**
 *
 * @author aymen
 */
public class Maitenance {
    private int id;
    private int id_u;
    private float prix;
    private String problem;
    private String etat;
    private String image;
    private String type;
    private Date DATE_D;
    private Date DATE_F;

    public Maitenance(float prix, String etat) {
        this.prix = prix;
        this.etat = etat;
    }

    public Maitenance(int id_u, float prix, String problem, String etat, String image, String type, Date DATE_D, Date DATE_F) {
        this.id_u = id_u;
        this.prix = prix;
        this.problem = problem;
        this.etat = etat;
        this.image = image;
        this.type = type;
        this.DATE_D = DATE_D;
        this.DATE_F = DATE_F;
    }

    public Maitenance(int id, int id_u, float prix, String problem, String etat, String image, String type, Date DATE_D, Date DATE_F) {
        this.id = id;
        this.id_u = id_u;
        this.prix = prix;
        this.problem = problem;
        this.etat = etat;
        this.image = image;
        this.type = type;
        this.DATE_D = DATE_D;
        this.DATE_F = DATE_F;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getId_u() {
        return id_u;
    }

    public void setId_u(int id_u) {
        this.id_u = id_u;
    }

    public float getPrix() {
        return prix;
    }

    public void setPrix(float prix) {
        this.prix = prix;
    }

    public String getProblem() {
        return problem;
    }

    public void setProblem(String problem) {
        this.problem = problem;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Date getDATE_D() {
        return DATE_D;
    }

    public void setDATE_D(Date DATE_D) {
        this.DATE_D = DATE_D;
    }

    public Date getDATE_F() {
        return DATE_F;
    }

    public void setDATE_F(Date DATE_F) {
        this.DATE_F = DATE_F;
    }
    
    
}

File: symfony/bike/src/MaitenanceBundle/Entity/maitenance.php
<?php

namespace MaitenanceBundle\Entity;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\ORM\Mapping as ORM;

/**
 * maitenance
 *
 * @ORM\Table(name="maitenance", indexes={@ORM\Index(name="FK_id_u", columns={"ID_U"})})
 * @ORM\Entity(repositoryClass="MaitenanceBundle\Repository\maitenanceRepository")
 */
class maitenance
{
    /**
     * @var int
     *
     * @ORM\Column(name="ID_M", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $IDM;
    /**
     * @var integer
     *
     * @ORM\Column(name="ID_U", type="integer")
     */
    private $idU;
    /**
     * @var string
     * @Assert\NotBlank
     * @Assert\Length(min=10)
     * @ORM\Column(name="Problem", type="text", length=65535, nullable=false)
     */
    private $Problem;
    /**
     * @var float
     *
     * @Assert\Range(
     *      min = 1,
     *      max = 999,
     *      minMessage = "Nombre petit",
     *      maxMessage = "Nombre grand"
     * )
     * @ORM\Column(name="Prix", type="float", precision=10, scale=0, nullable=false)
     */
    private $prix;
    /**
     * @var string
     *
     * @ORM\Column(name="etat", type="string", length=255, nullable=false)
     */
    private $etat;

    /**
     * @return string
     */
    public function getType()
    {
        return $this->type;
    }

    /**
     * @param string $type
     */
    public function setType(string $type)
    {
        $this->type = $type;
    }
    /**
     * @var string
     *
     * @ORM\Column(name="type", type="string", length=255, nullable=false)
     */
    private $type;
    /**
     * @var string
     *
     * @ORM\Column(name="image", type="string", length=255, nullable=false)
     */
    private $image;
    /**
     * @var \DateTime
     *
     * @ORM\Column(name="DATE_D", type="datetime", nullable=false, columnDefinition="DATETIME on update CURRENT_TIMESTAMP")
     */
    private $dateD;
    /**
     * @var \DateTime
     *
     * @ORM\Column(name="DATE_F", type="datetime")
     */
    private $dateF;

    /**
     * @return int
     */
    public function getIDM(): int
    {
        return $this->IDM;
    }

    /**
     * @param int $IDM
     */
    public function setIDM(int $IDM)
    {
        $this->IDM = $IDM;
    }





    /**
     * @return int
     */
    public function getIdU()
    {
        return $this->idU;
    }

    /**
     * @param int $idU
     */
    public function setIdU(int $idU)
    {
        $this->idU = $idU;
    }

    /**
     * @return string
     */
    public function getProblem()
    {
        return $this->Problem;
    }

    /**
     * @param string $Problem
     */
    public function setProblem(string $Problem)
    {
        $this->Problem = $Problem;
    }

    /**
     * @return float
     */
    public function getPrix()
    {
        return $this->prix;
    }

    /**
     * @param float $prix
     */
    public function setPrix(float $prix)
    {
        $this->prix = $prix;
    }

    /**
     * @return string
     */
    public function getEtat()
    {
        return $this->etat;
    }

    /**
     * @param string $etat
     */
    public function setEtat(string $etat)
    {
        $this->etat = $etat;
    }

    /**
     * @return mixed
     */
    public function getImage()
    {
        return $this->image;
    }

    /**
     * @param mixed $image
     */
    public function setImage(string $image)
    {
        $this->image = $image;
    }

    /**
     * @return \DateTime
     */
    public function getDateD()
    {
        return $this->dateD;
    }

    /**
     * @param \DateTime $dateD
     */
    public function setDateD(\DateTime $dateD)
    {
        $this->dateD = $dateD;
    }

    /**
     * @return \DateTime
     */
    public function getDateF()
    {
        return $this->dateF;
    }

    /**
     * @param \DateTime $dateF
     */
    public function setDateF(\DateTime $dateF)
    {
        $this->dateF = $dateF;
    }

}

